### PR TITLE
Show student's email on instructorCourseDetailsPage #4058

### LINKED ISSUE #4058: Show student's email on instructorCourseDetailsPage
The csv file for course details shows the email, so the page should show it too.

Page:
coursedetailspage

Csv:
coursedetailscsv

### PR DESCRIPTION
**Pull request by kennho**
Fixes #4058

Email column wasn't showing previously:

screen shot 2016-02-02 at 1 19 06 am

Updated the email column to show, which matches the csv downloaded:

screen shot 2016-02-02 at 1 18 24 am

screen shot 2016-02-02 at 1 19 34 am

### CODE REVIEW

**Comment by damithc** [comment]
@kennho is this ready for review? You haven't updated the tests yet.
In what other places we have student list? May be we can show the email in all those places, avoiding the need for conditional code.

**Comment by junhaoyap** [comment]
^ Was about to say please include tests so this won't regress in the future

File: src/main/webapp/WEB-INF/tags/instructor/studentList.tag [hunk 1]
```diff
@@ -31,16 +31,21 @@
...
-                        <th id="button_sortstudentstatus" class="button-sort-none" onclick="toggleSort(this,5)">
-                            Status <span class="icon-sort unsorted"></span>
-                        </th>
-                    </c:if>
+                    <c:choose>
+                        <c:when test="${fromCourseDetailsPage}">
+                            <th id="button_sortstudentstatus" class="button-sort-none" onclick="toggleSort(this,5)">
+                                Status <span class="icon-sort unsorted"></span>
+                            </th>
+                            <th id="button_sortemail-${courseIndex}" class="button-sort-none" onclick="toggleSort(this,6)">
```
Comment (junhaoyap) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: Not related to your changes, but I don't like how ```toggleSort``` takes in a magic number, very fragile, @damithc perhaps we should open an issue to look into how we can improve this function?

Comment (damithc) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: Yes, that's a good idea.

Comment (junhaoyap) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: @kennho could you look into opening your first issue - look into how to improve the ```toggleSort``` function to improve it - opening it doesn't mean you have to be the one to handle the issue

**Comment by kennho** [comment]
Updated the tests and opened the new issue. 

As for the email, it is already showing in all the places that uses student list. The conditional code is used to show / hide the Status column which will only be shown in the instructorCourseDetailsPage. It is also to ensure that the columns appear in the same order as it is in the csv downloaded. 

The conditional statement for the order of the columns can be removed if the order of the columns appearing in the csv is changed from Status | Email to Email | Status. However, the conditional statement to hide the Status column will still be there.

Otherwise, we can just show the status column in all the places that uses the student list. This will remove the conditional statement completely. Pages affected by the student list will be instructorCourseDetailsPage, instructorStudentListPage and instructorSearchPage. 

The Status column looks relavent to be added in the rest of the pages mentioned. So maybe we can just remove the conditional code and allow the Status column to be seen in every student list? However I am not sure if there are any reason for not showing the Status column in the instructorStudentListPage and instructorSearchPage from the start.

@damithc @junhaoyap

**Comment by junhaoyap** [comment]
@kennho good job on opening your first issue! :)

Your current changes are fine, no worries

I'll be flying back to Singapore so I might not respond for the next 2 days while it's being discussed

**Comment by junhaoyap** [comment]
However, I'm not sure about any reason for not showing the status column, perhaps @damithc can give you a better idea on whether that is ok? I'll add a ```s.ToDiscuss``` label until that is resolved

**Comment by wkurniawan07** [comment]
I'll chip in here, this has something to do with the JSP -> JSTL migration some time back.

Prior to the migration, the web pages are rendered using pure JSP which has very limited capability of component reuse (only minor components such as header/footer are possible). As such, many similar/identical components have to be rendered separately. A glaring violation of DRY, but it's a limitation imposed by JSP. Because of that, naturally there will be minor/major variations in components intended to be similar/identical, e.g forgetting to copy/paste certain parts :p (I think some senior developers can account on the troubles I had to go through back then to standardize 10+ variations of how comments are rendered in different places :p)

Back to this issue on status column, I have a high confidence that it's ok to include because it's just another case of "forget-to-copy-and-paste", but let's wait for more inputs.

**Comment by damithc** [comment]
At some point we introduced 'submit without joining' feature. Since then, the join status is no longer important, so may be we should not waste space by showing it everywhere. However, I'm ok to show it everywhere in the interest of simpler code. If the space becomes a problem in future, we can revisit this issue.

**Comment by kennho** [comment]
I have updated the code to show the Status column everywhere. Updated the tests too.

**Comment by junhaoyap** [comment]
Acknowledged, I'm currently transiting in JP so I won't be able to review, I'll review when I'm settled back in

**Comment by thyageshm** [comment]
@junhaoyap @kennho please try and complete this one by this release

**Comment by junhaoyap** [comment]
Running the tests now, will update in a few moments

**Comment by junhaoyap** [comment]
Setting to merge - @kennho do remember to merge from teammates master again, some updates are required, @damithc please do the final review for merge approval

**Comment by kennho** [comment]
merged from the master

**Comment by junhaoyap** [comment]
Awesome, I will merge it after dinner and some errands later tonight, good work @kennho

**Comment by junhaoyap** [comment]
@kennho merged, again, great job :+1:

**Pull request merged**